File: scripts/generate-catalog.js
#!/usr/bin/env node

/**
 * Generate CATALOG.md from skills_index.json
 * Auto-generates a human-readable catalog of all skills
 */

const fs = require('fs');
const path = require('path');

// Read skills index
const indexPath = path.join(__dirname, '../skills_index.json');
if (!fs.existsSync(indexPath)) {
  console.error('❌ skills_index.json not found. Run generate-skills-index.js first.');
  process.exit(1);
}

const index = JSON.parse(fs.readFileSync(indexPath, 'utf8'));

// Start building catalog
let catalog = `# 📚 CLI AI Skills Catalog

**Generated:** ${new Date().toISOString()}  
**Total Skills:** ${index.skills.length}  
**Platforms:** GitHub Copilot CLI, Claude Code, OpenAI Codex

---

## 📋 All Skills

`;

// Add skills table
catalog += `| Skill | Version | Category | Tags | Risk | Platforms |
|-------|---------|----------|------|------|-----------|
`;

index.skills.forEach(skill => {
  const tags = (skill.tags || []).slice(0, 3).join(', ');
  const platforms = (skill.platforms || []).map(p => {
    if (p === 'github-copilot-cli') return '🤖';
    if (p === 'claude-code') return '🧠';
    if (p === 'codex') return '⚙️';
    return p;
  }).join(' ');
  
  catalog += `| **${skill.name}** | ${skill.version} | ${skill.category} | ${tags}... | ${skill.risk} | ${platforms} |
`;
});

catalog += `

---

## 🎯 Skills by Category

`;

// Group by category
const categories = {};
index.skills.forEach(skill => {
  if (!categories[skill.category]) {
    categories[skill.category] = [];
  }
  categories[skill.category].push(skill);
});

Object.entries(categories).forEach(([category, skills]) => {
  catalog += `### ${category.charAt(0).toUpperCase() + category.slice(1)}\n\n`;
  
  skills.forEach(skill => {
    catalog += `- **${skill.name}** (v${skill.version})\n`;
    catalog += `  - Description: ${skill.description.substring(0, 100)}...\n`;
    catalog += `  - Tags: ${(skill.tags || []).join(', ')}\n\n`;
  });
});

catalog += `---

## 🏷️ Skills by Tag

`;

// Group by tags
const tags = {};
index.skills.forEach(skill => {
  (skill.tags || []).forEach(tag => {
    if (!tags[tag]) {
      tags[tag] = [];
    }
    tags[tag].push(skill);
  });
});

Object.entries(tags)
  .sort((a, b) => b[1].length - a[1].length)
  .slice(0, 20) // Top 20 tags
  .forEach(([tag, skills]) => {
    catalog += `- **${tag}** (${skills.length}): ${skills.map(s => s.name).join(', ')}\n`;
  });

catalog += `

---

## 📦 Curated Bundles

See [Bundles Guide](docs/bundles/bundles.md) for curated collections of skills by use case.

`;

// Write catalog
const catalogPath = path.join(__dirname, '../CATALOG.md');
fs.writeFileSync(catalogPath, catalog);

console.log(`✅ CATALOG.md generated (${index.skills.length} skills)`);
console.log(`📄 File: ${catalogPath}`);

File: scripts/generate-catalog.py
#!/usr/bin/env python3
"""Generate CATALOG.md from skills_index.json"""

import json
from datetime import datetime
from pathlib import Path

# Read skills index
index_path = Path(__file__).parent.parent / 'skills_index.json'
with open(index_path) as f:
    index = json.load(f)

# Start building catalog
catalog = f"""# 📚 Claude Superskills Catalog

**Generated:** {datetime.now().isoformat()}Z  
**Total Skills:** {len(index['skills'])}  
**Platforms:** GitHub Copilot CLI, Claude Code, OpenAI Codex, OpenCode, Gemini CLI, Antigravity, Cursor IDE, AdaL CLI

---

## 📋 All Skills

| Skill | Version | Category | Tags | Risk | Platforms |
|-------|---------|----------|------|------|-----------|
"""

# Add skills table
for skill in index['skills']:
    tags = ', '.join(skill['tags'][:3]) if skill['tags'] else 'N/A'
    platforms = ' '.join({
        'github-copilot-cli': '🤖',
        'claude-code': '🧠',
        'codex': '⚙️',
        'opencode': '🧩',
        'gemini': '♊',
        'antigravity': '🪂',
        'cursor': '🖱️',
        'adal': '🧪'
    }.get(p, p) for p in skill['platforms'])
    
    catalog += f"| **{skill['name']}** | {skill['version']} | {skill['category']} | {tags} | {skill['risk']} | {platforms} |\n"

catalog += "\n---\n\n## 🎯 Skills by Category\n\n"

# Group by category
categories = {}
for skill in index['skills']:
    cat = skill['category']
    if cat not in categories:
        categories[cat] = []
    categories[cat].append(skill)

for category in sorted(categories.keys()):
    catalog += f"### {category.capitalize()}\n\n"
    for skill in categories[category]:
        desc_preview = skill['description'][:100] if skill['description'] else 'N/A'
        catalog += f"- **{skill['name']}** (v{skill['version']})\n"
        catalog += f"  - Description: {desc_preview}...\n"
        catalog += f"  - Tags: {', '.join(skill['tags'])}\n\n"

catalog += "\n---\n\n## 📦 Curated Bundles\n\n"
catalog += "See [Bundles Guide](docs/bundles/bundles.md) for curated collections of skills by use case.\n"

# Write catalog
catalog_path = Path(__file__).parent.parent / 'CATALOG.md'
with open(catalog_path, 'w') as f:
    f.write(catalog)

print(f"✅ CATALOG.md generated ({len(index['skills'])} skills)")
